VMware: move vmware module_utils

Akasurde opened the pull request:
##### SUMMARY
vmware.py from module_utils is moved to vmware_common directory
as vmware namespace is colliding with other VMware official
libraries e.g. vSphere Automation SDK.
vmware.py has new home under vmware_common.

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
lots of them

##### ANSIBLE VERSION
<!--- Paste verbatim output from "ansible --version" between quotes below -->
```
2.6devel
```

Akasurde commented:
@tchernomax @pdellaert @dagwieers @jctanner Could you please review this ? This change is required as I am introducing new vmware module_utils for REST calls which is based upon vSphere Automation SDK. There is clash between vmware.py and vSphere Automation SDK namespaces, so moving vmware.py in vmware_common directory. Please refer this [vmware_tag](https://github.com/ansible/ansible/pull/37261) PR for more details.

Akasurde commented:
Can someone review this ?

gundalow commented:
You may wish to add this new directory into BOTMETA.yml as well

gundalow commented:
BOTMETA.yml for maintainers & to add vmware label

pdellaert reviewed (approved):
LGTM

webknjaz commented on lib/ansible/module_utils/vmware_common/vmware.py:
```diff
@@ -0,0 +1,1195 @@
+            except AttributeError:
+                raise TaskError("An unknown error has occurred")
+        if task.info.state == vim.TaskInfo.State.running:
+            time.sleep(15)
```
How about doing a [truncated exponential backoff](https://cloud.google.com/storage/docs/exponential-backoff) here?

Akasurde replied:
@webknjaz This is a legacy code and I will address this in different PR.

Akasurde committed:
VMware: move vmware module_utils

vmware.py from module_utils is moved to vmware_common directory
as vmware namespace is colliding with other VMware official
libraries e.g. vSphere Automation SDK.
vmware.py has new home under vmware_common.

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Akasurde commented:
@gundalow Done.

abadger commented:
How long has module_utils/VMware.py existed?  Maybe you need to include a
VMware.py at the old location for backwards compatiblity.  (Import all the
public symbols from the new location.  It would be good to return a
deprecaution warning as well but you won't have access to an ansible module
object when it's imported so I don't know that there's a way to do that....

abadger commented:
Why not module_utils/vmware/common.py ?

mattclay commented:
@Akasurde Can you provide an example of what the conflict is?

Akasurde commented:
Closing this PR as import boilerplate fixes the import conflicts. Thanks everyone for reviews and comments.
